### ISSUE TITLE
Steal ping-ponging

### ISSUE THREAD

**Issue by GoogleCodeExporter** [j-woz/exm-issues#360]
```
I've come across a pathological situation where two ADLB servers spend almost 
all of their time stealing work from one another.  It seems to occur when the 
work queues get quite full.

Here's an example of it happening on the loop benchmark

I run this:
  export ADLB_DEBUG=1; export ADLB_SERVERS=2; export TURBINE_ENGINES=2; time turbine -n6 589-big-loop.tcl -bound=10000000 -sleeptime=0 > /var/tmp/steal.log

If I look in top I see the two ADLB server processes consuming 100% CPU and the 
other processes mostly idle.  I  terminated it after about two minutes and 
grepped the logs:

[/var/tmp]$ grep 'ADLB_Get' /var/tmp/steal.log | wc -l                          

39585
[/var/tmp]$ grep 'ADLB_Put' /var/tmp/steal.log  | wc -l                         

99236
[/var/tmp]$ grep 'handle.*STEAL' /var/tmp/steal.log  | wc -l                    

12881
[/var/tmp]$ grep 'stolen' /var/tmp/steal.log  | wc -l                           

26525688

Basically it looks like the same tasks are being repeatedly swapped back and 
forth, while very little real work is being done.
```

Original issue reported on code.google.com by `tim.g.ar...@gmail.com` on 29 Nov 2012 at 11:35
